contracts/SSVNetwork.sol

```diff
@@ -58,32 +71,76 @@ contract SSVNetwork is OwnableUpgradeable, ISSVNetwork {
     mapping(bytes32 => Cluster) private _clusters;
     mapping(bytes32 => Pod) private _pods;
     mapping(bytes32 => Validator) _validatorPKs;
 
     uint64 private _networkFee;
+    uint64 private _networkFeeIndex;
+    uint64 private _networkFeeIndexBlockNumber;
+
+    uint64 private _declareOperatorFeePeriod;
+    uint64 private _executeOperatorFeePeriod;
+    uint64 private _operatorMaxFeeIncrease;
+
     uint64 constant LIQUIDATION_MIN_BLOCKS = 50;
+    uint64 constant MINIMAL_OPERATOR_FEE = 100000000;
+
     // uint64 constant NETWORK_FEE_PER_BLOCK = 1;
 
 
     DAO private _dao;
     IERC20 private _token;
 
-    function initialize(IERC20 token_) external initializer override {
+    function initialize(
+        IERC20 token_,
+        uint64 operatorMaxFeeIncrease_,
+        uint64 declareOperatorFeePeriod_,
+        uint64 executeOperatorFeePeriod_
+    ) external override {
+        __SSVNetwork_init(token_, operatorMaxFeeIncrease_, declareOperatorFeePeriod_, executeOperatorFeePeriod_);
+    }
+
+    function __SSVNetwork_init(
+        IERC20 token_,
+        uint64 operatorMaxFeeIncrease_,
+        uint64 declareOperatorFeePeriod_,
+        uint64 executeOperatorFeePeriod_
+    ) internal initializer {
+        __Ownable_init_unchained();
+        __SSVNetwork_init_unchained(token_, operatorMaxFeeIncrease_, declareOperatorFeePeriod_, executeOperatorFeePeriod_);
+    }
+
+    function __SSVNetwork_init_unchained(
+        IERC20 token_,
+        uint64 operatorMaxFeeIncrease_,
+        uint64 declareOperatorFeePeriod_,
+        uint64 executeOperatorFeePeriod_
+    ) internal onlyInitializing {
         _token = token_;
+        _updateOperatorFeeIncreaseLimit(operatorMaxFeeIncrease_);
+        _updateDeclareOperatorFeePeriod(declareOperatorFeePeriod_);
+        _updateExecuteOperatorFeePeriod(executeOperatorFeePeriod_);
+    }
+
+    modifier onlyOperatorOwnerOrContractOwner(uint64 operatorId) {
+        _onlyOperatorOwnerOrContractOwner(operatorId);
+        _;
     }
 
     function registerOperator(
         bytes calldata encryptionPK,
-        uint64 fee
+        uint256 fee
     ) external returns (uint64 id) {
-        if (fee <= 0) revert FeeTooLow();
+        if (fee < MINIMAL_OPERATOR_FEE) {
+            revert FeeTooLow();
+        }
 
         lastOperatorId.increment();
         id = uint64(lastOperatorId.current());
-        _operators[id] = Operator({ owner: msg.sender, snapshot: Snapshot({ block: uint64(block.number), index: 0, balance: 0}), validatorCount: 0, fee: fee});
-        emit OperatorAdded(id, msg.sender, encryptionPK);
+        _operators[id] = Operator({ owner: msg.sender, snapshot: Snapshot({ block: uint64(block.number), index: 0, balance: 0}), validatorCount: 0, fee: fee.shrink()});
+        emit OperatorAdded(id, msg.sender, encryptionPK, fee.shrinkable());
     }
 
     function removeOperator(uint64 operatorId) external {
```

Comment:
why cant we just remove that operator instead of setting all to 0?
cause if we want to re-register it will get new id?

Reply:
also, we need to check that it doesn't have a balance, no? or maybe include on remove it also withdraw

Reply:
@lior-blox that was exists in v3. the sense of pr is to show the diff what was done for last month. If we are talking about business logic, then we have a gap. If we remove operator what we should do with clusters?